File: backend/src/main/java/com/printit/backend/features/notifications/NotificationResponse.java
package com.printit.backend.features.notifications;

import java.time.LocalDateTime;

public class NotificationResponse {

    private Long id;
    private String title;
    private String message;
    private String type;
    private boolean isRead;
    private LocalDateTime createdAt;

    public NotificationResponse(
            Long id,
            String title,
            String message,
            String type,
            boolean isRead,
            LocalDateTime createdAt
    ) {
        this.id = id;
        this.title = title;
        this.message = message;
        this.type = type;
        this.isRead = isRead;
        this.createdAt = createdAt;
    }

    public Long getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public String getMessage() {
        return message;
    }

    public String getType() {
        return type;
    }

    public boolean isRead() {
        return isRead;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }
}

File: mobile/app/src/main/java/com/printit/mobile/features/notifications/NotificationResponse.kt
package com.printit.mobile.features.notifications

import com.google.gson.annotations.SerializedName

data class NotificationResponse(
    @SerializedName("id")
    val id: Long?,

    @SerializedName("title")
    val title: String?,

    @SerializedName("message")
    val message: String?,

    @SerializedName("type")
    val type: String?,

    @SerializedName("isRead")
    val isRead: Boolean?,

    @SerializedName("createdAt")
    val createdAt: String?
)